### PR TITLE
Exclude version update code (fixes #112) and dotfiles from release tarballs.

### LINKED ISSUE #112: setup.py overwrites nml/__version__.py
Another problem cropped up in the Debian package: Running `./setup.py` overwrites the `nml/__version__.py` supplied in the source tarball.

Normally, this is ok, since it gets the nml tag or git hash that is being compiled. However, my Debian package also use git, so when I try to build the .deb from my git repo, the __version__.py gets overwritten with an incorrect git hash or tag.

I tried adding a Debian-specific patch to prevent this, but typically a Debian package is cleaned before applying patches, so then the version number is already gone, so this should really be fixed in the tarball.

I wonder if the release tarball could somehow know that it is a release tarball and prevent its __version__.py from being overwritten? Alternatively, passing some option to setup.py or setting some environment variable that would prevent the git version detection and forces using the existing `__version__.py` would also work fine.

### PR DESCRIPTION
**Pull request by FLHerne**
Closes #115

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -4,9 +4,14 @@
 
 from setuptools import setup, Extension, find_packages
 from setuptools.command.build_py import build_py
-from nml import version_info
-NML_VERSION = version_info.get_and_write_version()
 
+try:
+    from nml import version_update
+    NML_VERSION = version_update.get_and_write_version()
+except ImportError:
```
Comment (LordAro) [inline setup.py:11]: in normal usage, version_info would be used, not version_update, so perhaps that should go first?

Comment (glx22) [reply setup.py:11]: When you create a bundle you want to update the version first. When ran from a bundle it will silently fall back to version_info.

Comment (FLHerne) [reply setup.py:11]: What @glx22 says.

**Comment by matthijskooijman** [comment]
I'm a bit late to the party, but these changes look good to me! Given `setuptools_scm` wasn't a clear all-in-one solution, it could have been used just as as library to parse git tags (and maybe PKG_INFO files), but since we already had code for that, probably better to not add one more dependency. Thanks for working on this!

**Pull request merged**